### PR TITLE
Add counting leaderboard

### PR DESCRIPTION
**Pull request by cjdenio**
This PR adds a slash command (available at http://server/slack/leaderboard) that shows monthly counting stats!

### CODE REVIEW

**Review by khrj — APPROVED** [review]

**Comment by ifvictr** [comment]
@cjdenio This is awesome! Do you mind adding a screenshot to this thread so we can get a preview?

**Comment by cjdenio** [comment]
Of course! Here's how it works:

image

File: pkg/jia/handlers.go [hunk 1]
```diff
@@ -80,4 +83,84 @@ func onMessage(slackClient *slack.Client, event *slackevents.MessageEvent) {
...
 	redisClient.Set("last_valid_number", matchedNumber, 0)
 	redisClient.Set("last_sender_id", event.User, 0)
+
+	// Get the current month/year in UTC
+	now := time.Now().UTC()
+	year := now.Year()
+	month := now.Month()
+
+	// Increment the person's monthly count
+	redisClient.Incr(fmt.Sprintf("leaderboard:%d-%d:%s", month, year, event.User))
```
Comment (ifvictr) [inline pkg/jia/handlers.go:93]: Can you change it so that it's YYYY-DD instead?

File: pkg/jia/handlers.go [hunk 2]
```diff
@@ -80,4 +83,84 @@ func onMessage(slackClient *slack.Client, event *slackevents.MessageEvent) {
...
+}
+
+func HandleLeaderboardSlashCommand(w http.ResponseWriter, r *http.Request) {
+	// Get the current month/year in UTC
+	now := time.Now().UTC()
+	year := now.Year()
+	month := now.Month()
+
+	scan := redisClient.Scan(0, fmt.Sprintf("leaderboard:%d-%d:*", month, year), 10)
```
Comment (ifvictr) [inline pkg/jia/handlers.go:102]: Same here

File: pkg/jia/handlers.go [hunk 3]
```diff
@@ -80,4 +83,84 @@ func onMessage(slackClient *slack.Client, event *slackevents.MessageEvent) {
...
+	if scan.Err() != nil {
+		w.Write([]byte("Something went wrong while loading the leaderboard :cry: Please try again later!"))
+		return
+	}
+
+	scan_iterator := scan.Iterator()
```
Comment (ifvictr) [inline pkg/jia/handlers.go:108]: Use camel case :P

File: pkg/jia/handlers.go [hunk 4]
```diff
@@ -80,4 +83,84 @@ func onMessage(slackClient *slack.Client, event *slackevents.MessageEvent) {
...
+	type Entry struct {
+		Number int
+		User   string
+	}
+
+	entries := []Entry{}
+
+	for scan_iterator.Next() {
+		entry := redisClient.Get(scan_iterator.Val())
+		entry_int, err := entry.Int()
```
Comment (ifvictr) [inline pkg/jia/handlers.go:119]: Same here!

**Review by ifvictr — CHANGES_REQUESTED** [review]

**Review by ifvictr — APPROVED** [review]

**Comment by ifvictr** [comment]
Thanks! This feature will be live soon.

**Pull request merged**